packages/flutter_tools/lib/src/base/net.dart

```diff
@@ -29,6 +29,33 @@ Future<List<int>> fetchUrl(Uri url) async {
   }
 }
 
+Future<bool> doesRemoteFileExist(Uri url) async {
```

Comment:
Good point. I'm not that concerned about the retry logic since this is only trying to check for a valid response from the server, but the majority of the code is shareable.